src/dotnet/TransactionResourceManager.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Transactions;
+
+namespace Microsoft.DotNet.Cli
+{
+    public class TransactionResourceManager : IEnlistmentNotification
```

Comment:
This name is confusing to me as I don't think it is a "manager" since it isn't managing resources; fundamentally, it's a generic transaction enlistment mechanism.

Reply:
I was wonder about that for a well. I hate "XXmanager" as well. But "ResourceManger" is a specific term that means object implements IEnlistmentNotification. Maybe I will just call it "TransactionEnlistment" ?

Reply:
Possible suggestion of `TransactionNotification`?  I'd like to see the `EnlistVolatile` call encapsulated, but that's just my personal preference; I'm also fine with having the users of this type having to explicitly call `EnlistVolatile`.

Reply:
I'm good with `TransactionEnlistment`.

Reply:
ToolPackageObtainTransaction and CreateShimTransaction